### PR TITLE
Use password strength when setting a new password

### LINKED ISSUE #169: Use a password-strength library for the password indicator

### PR DESCRIPTION
**Pull request by sisou**
New passwords still require at least 8 characters, but then also require a password score above 40(%). Only then is the button displayed and the form can be submitted.

Resolves #169.

### CODE REVIEW

File: src/components/PassphraseSetterBox.js [hunk 1]
```diff
@@ -43,9 +44,10 @@ class PassphraseSetterBox extends Nimiq.Observable {
         /* eslint-disable max-len */
         $el.innerHTML = `
             <div class="password-strength strength-0  nq-text-s" data-i18n="passphrasebox-password-strength-0" >Enter at least 8 characters</div>
-            <div class="password-strength strength-8  nq-text-s" data-i18n="passphrasebox-password-strength-8" >Great, that's a good password!</div>
-            <div class="password-strength strength-10 nq-text-s" data-i18n="passphrasebox-password-strength-10">Super, that's a strong password!</div>
-            <div class="password-strength strength-12 nq-text-s" data-i18n="passphrasebox-password-strength-12">Excellent, that's a very strong password!</div>
+            <div class="password-strength strength-4  nq-text-s" data-i18n="passphrasebox-password-strength-4" >Hm, that's a weak password :(</div>
```
Comment (nibhar) [inline src/components/PassphraseSetterBox.js]: `hm`, `:(`? 
```suggestion
            <div class="password-strength strength-4  nq-text-s" data-i18n="passphrasebox-password-strength-4" >That is a weak password!</div>
```
I suggest using `that is` over `that's`. I also suggest using punctuation evenly (including `strength-0`).

Comment (sisou) [reply src/components/PassphraseSetterBox.js]: I now adapted to your wording suggestions, however we will have to go over all wording together with Julian and Tammo anyway again.

**Comment by sisou** [comment]
Rebased on master because of merge conflict.

File: src/components/PassphraseSetterBox.js [hunk 2]
```diff
@@ -105,25 +113,34 @@ class PassphraseSetterBox extends Nimiq.Observable {
...
-        this.$el.classList.toggle('strength-12', length >= 12);
+        const score = PasswordStrength.strength(this._passphraseInput.text);
+
+        this.$el.classList.toggle('input-eligable', isValid && score >= 40);
+
+        this.$el.classList.toggle('strength-short', !isValid);
+        this.$el.classList.toggle('strength-weak', isValid && score < 40);
+        this.$el.classList.toggle('strength-good', isValid && score >= 40 && score < 75);
+        this.$el.classList.toggle('strength-strong', isValid && score >= 75 && score < 150);
+        this.$el.classList.toggle('strength-secure', isValid && score >= 150);
```
Comment (Bettelstab) [inline src/components/PassphraseSetterBox.js]: I would prefer those numbers to be constants.

Comment (Bettelstab) [inline src/components/PassphraseSetterBox.js]: I would prefer those numbers to be constants.

File: src/components/PassphraseSetterBox.js [hunk 3]
```diff
@@ -105,25 +113,34 @@ class PassphraseSetterBox extends Nimiq.Observable {
...
+    }
+
+    _isEligablePassword() {
+        const password = this._passphraseInput.text;
+        if (password.length < PassphraseInput.DEFAULT_MIN_LENGTH) return false;
+        return PasswordStrength.strength(password) >= 40;
```
Comment (Bettelstab) [inline src/components/PassphraseSetterBox.js]: constant

**Pull request merged**